src/build.js

```diff
@@ -82,6 +82,29 @@ export const build = function(statics) {
         } else if ((!buffer && char === '{') || char === '}') {
           // First `{` after opening expression `{{`.
           expr = EXPR_RAW;
+        } else if ((!buffer && char === '!')) {
+          const isSpecialComment = str.substr(j, 3) === '!--';
+          // Consume the entire comment
+          for (; j < str.length; j++) {
```

Comment:
we should try to avoid inner loops and just let the one we have run it's course.

Reply:
That was my initial take as well, however, it would require adding more states and then basically adding if-cases checking if we're currently in a comment and then just ignore the content. I therefore opted for the inner loop, consuming the content instead. That way the feature was more centralized.

Reply:
I'll rewrite it to rely on the main loop instead.

Reply:
that's true. it would require more states but should be less code overall I think.

Reply:
I've pushed some changes to rely on the outer loop. I'm unsure of if it's done in a way sensible to how the parser is intended to function in regards to state.

The issue with `undefined` being in `children` still persists. How do I add an empty string after the end of the HTML-tag following the mustache?

Take `<div>{{!Comment }}</div>` for example, it currently yields `[['<div>', '</div>']`. Is there a missing `commit()` needed to be done, and if so where?

PS: If you want me to clean up the commits by squashing them when the functionality is there, please let me know.